Upstream-Status: Submitted [https://github.com/projectacrn/acrn-kernel/compare/acrn_skl_test]
Signed-off-by: Chin Huat Ang <chin.huat.ang@intel.com>

From 6bb3b97621b0bf41cfc10df807984d1b0c17a01e Mon Sep 17 00:00:00 2001
From: He Min <min.he@intel.com>
Date: Fri, 8 Mar 2019 08:13:08 +0000
Subject: [PATCH 5/8] drm/i915/gvt: handles error when ioreq attach client
 fails

In case of device model crashes accidentally, VHM will not be
able to attach ioreq client, so acrngt needs to handle this
failure case.

Signed-off-by: He Min <min.he@intel.com>
Signed-off-by: Fengwei Yin <fengwei.yin@intel.com>
Reviewed-by: Zhao Yakui <yakui.zhao@intel.com>
---
 drivers/gpu/drm/i915/gvt/acrngt.c | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/gvt/acrngt.c b/drivers/gpu/drm/i915/gvt/acrngt.c
index 0f1154c..e27b53b 100644
--- a/drivers/gpu/drm/i915/gvt/acrngt.c
+++ b/drivers/gpu/drm/i915/gvt/acrngt.c
@@ -237,7 +237,14 @@ static int acrngt_emulation_thread(void *priv)
 
 	set_freezable();
 	while (1) {
-		acrn_ioreq_attach_client(info->client, 1);
+		ret = acrn_ioreq_attach_client(info->client, 1);
+
+		if (ret) {
+			gvt_err("error while attach ioreq client %d\n", ret);
+			info->client = 0;
+			info->emulation_thread = NULL;
+			return 0;
+		}
 
 		if (kthread_should_stop())
 			return 0;
-- 
2.7.4

